mailbox/scanning-search/src/test/java/org/apache/james/mailbox/store/search/PDFTextExtractor.java

```diff
@@ -0,0 +1,63 @@
+/****************************************************************
+ * Licensed to the Apache Software Foundation (ASF) under one   *
+ * or more contributor license agreements.  See the NOTICE file *
+ * distributed with this work for additional information        *
+ * regarding copyright ownership.  The ASF licenses this file   *
+ * to you under the Apache License, Version 2.0 (the            *
+ * "License"); you may not use this file except in compliance   *
+ * with the License.  You may obtain a copy of the License at   *
+ *                                                              *
+ *   http://www.apache.org/licenses/LICENSE-2.0                 *
+ *                                                              *
+ * Unless required by applicable law or agreed to in writing,   *
+ * software distributed under the License is distributed on an  *
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY       *
+ * KIND, either express or implied.  See the License for the    *
+ * specific language governing permissions and limitations      *
+ * under the License.                                           *
+ ****************************************************************/
+package org.apache.james.mailbox.store.search;
+
+import java.io.IOException;
+import java.io.InputStream;
+
+import org.apache.commons.io.IOUtils;
+import org.apache.james.mailbox.extractor.ParsedContent;
+import org.apache.james.mailbox.extractor.TextExtractor;
+import org.apache.pdfbox.pdmodel.PDDocument;
+import org.apache.pdfbox.text.PDFTextStripper;
+
+import com.google.common.base.Charsets;
+import com.google.common.collect.ImmutableMap;
+
+public class PDFTextExtractor implements TextExtractor {
+
+    private static final String PDF_TYPE = "application/pdf";
+
+    @Override
+    public ParsedContent extractContent(InputStream inputStream, String contentType) throws Exception {
+        if (isPDF(contentType)) {
+            return extractTextFromPDF(inputStream);
+        }
+        try {
+            return new ParsedContent(IOUtils.toString(inputStream, Charsets.UTF_8), ImmutableMap.of());
+        } catch (IOException e) {
+            return new ParsedContent(null, ImmutableMap.of());
```

Comment:
Well well well. Maybe here we don't have some parsed content. 

What about throwing an exception and let the client handle it? Would IMO seem better than lying to him.

Reply:
A quick look other the codebase confirms clients handle thrown exceptions.